### ISSUE TITLE
Conversion rate input for transfer type transaction advanced (checked)

### ISSUE THREAD

**Issue by reddyshyam** [moneymanagerex/moneymanagerex#2361]
Hi @vomikan 

Just a feature request please. Can we have the option of entering conversion rate when 'Advanced' option is checked in transaction type 'Transfer'. I believe it will be good if multi currency transfer is used which is why I assume 'Advanced' option is there. 

Either a new field or a toggle between rate or value might be a better option/approach. Please see if this makes any sense at all.

Illustration in the pic attached.
![Screenshot_04Apr2020_145817](https://user-images.githubusercontent.com/12392746/78423643-61195700-7685-11ea-953d-295dcef613fb.png)

**Comment by vomikan** [comment]
![image](https://user-images.githubusercontent.com/6836805/78445061-c8cda300-7681-11ea-9d7e-88f9b9f6f46d.png)

**Comment by reddyshyam** [comment]
Hi @vomikan, 

Thanks for the quick turn around of the idea. That might look handy but what about the conversion rate? From the implementation I can see/assume once the transaction is submitted or saved, the amount is calculated, pushed to respective column and saved. If conversion rate is provided, shouldn't we save it for future reference? With multi currency, another proper column to hold the conversation rate of the transaction might be very useful if one wants a report on conversion rate fluctuations. What do you think?

**Comment by vomikan** [comment]
There is a possibility to download historical rates for any currency.
Using this all saved transactions may be validated (by report for example).
IMHO no sense to multiply entered value to exchange rate, because real amount is using specific bank exchange rate + commission. Also account may be not selected yet.
But to have  info field with exchange rate for the entered transaction that good idea.

**Comment by reddyshyam** [comment]
> There is a possibility to download historical rates for any currency.
> Using this all saved transactions may be validated (by report for example).
> IMHO no sense to multiply entered value to exchange rate, because real amount is using specific bank exchange rate + commission. Also account may be not selected yet.
> But to have info field with exchange rate for the entered transaction that good idea.

Yes @vomikan. I am for a new field as thats the rate the amount was transferred which included bank charges, etc... and would not be the exchange rate of the day. So a new field will make sense and apt too. Please consider it.

**Comment by grandmotivator** [comment]
+

**Comment by whalley** [comment]
> Hi @vomikan,
> 
> From the implementation I can see/assume once the transaction is submitted or saved, the amount is calculated, pushed to respective column and saved. If conversion rate is provided, shouldn't we save it for future reference? 

Why would we need to save the rate? Can it not be calculated from the two amounts for advanced transfers? Or am I missing something here?

**Comment by reddyshyam** [comment]
> > From the implementation I can see/assume once the transaction is submitted or saved, the amount is calculated, pushed to respective column and saved. If conversion rate is provided, shouldn't we save it for future reference?
> 
> Why would we need to save the rate? Can it not be calculated from the two amounts for advanced transfers? Or am I missing something here?

yes we can but I, personally would love to see the exchange rate of the transactiion, recorded or calculated on the fly, in the transactions. When one has plenty of records, calculating manually isn't an easy way out but if displayed would be great to track.

**Comment by vomikan** [comment]
I agree that it would be desirable to display currency codes and the exchange rate in this dialog.
The only thing that stops is that you first need to select accounts to get this information. I still don't want to clutter up this dialog with information, or rather, I don't want to break its geometry.

**Comment by reddyshyam** [comment]
> I agree that it would be desirable to display currency codes and the exchange rate in this dialog. The only thing that stops is that you first need to select accounts to get this information. I still don't want to clutter up this dialog with information, or rather, I don't want to break its geometry.

This is a requirement I raised long back and forgotten too. I reason I wanted it in first first place was to record the rate on that date but as @whalley says, it can be calculated on the fly too. As long as it is displayed on the transaction history somehow, I am ok for this not to implemented too but I cannot speak for the rest (@grandmotivator  and others).

If everyone agrees, then this can be closed and new implementation of displaying rates involving multiple currencies taken up, I would be a happy bunny.

**Comment by whalley** [comment]
> When one has plenty of records, calculating manually isn't an easy way out but if displayed would be great to track.

So would it be more beneficial to include this in the transaction report?

**Comment by reddyshyam** [comment]
> > When one has plenty of records, calculating manually isn't an easy way out but if displayed would be great to track.
> 
> So would it be more beneficial to include this in the transaction report?

Yes, please.

**Comment by tactilis** [comment]
> I would suggest prepending the rate to the Notes field
> Views?

I'm fairly neutral on this Mark. It is a special case affecting a smallish number of people. The majority of users will have the same currency for their FROM and TO accounts so will never see it.

<br>
FYI, In general I'm against the idea of prepending text to users' Notes. It feels like a kludge. [1] 

I'd really like to see the '[Att]' go from Notes and be replaced by an Attachment column in the transaction list. I'll write a separate enhancement proposal for this at some point so that the pros and cons can be discussed. 


___

For non-native English speakers:
**[1]** A _kludge_ is:
- A workaround or quick-and-dirty solution that is clumsy, inelegant, inefficient, difficult to extend and hard to maintain.
- Any construction or practice, typically crude yet effective, designed to solve a problem temporarily or expediently.

**Comment by reddyshyam** [comment]
@whalley, I will agree with @tactilis. I do not want user notes changed for any reason without their knowledge. I am still inclined towards new column and if effects only a small set of users, then the user can make the column hidden if not required.

**Comment by n-stein** [comment]
> Rather than clutter the report by adding a new field to the transaction report

We could make it so the column only shows up if a transaction in the view has an exchange rate != 1. That would make it hidden for the majority of users, but visible to those few who use this feature.

**Comment by reddyshyam** [comment]
> > Rather than clutter the report by adding a new field to the transaction report
> 
> We could make it so the column only shows up if a transaction in the view has an exchange rate != 1. That would make it hidden for the majority of users, but visible to those few who use this feature.

Also user given choice to make the column hidden or visible please.

**Comment by whalley** [comment]
Actually thinking about it it's probably best to normally display the column but allow the user to specify whether they want to disable. This is then consistent with all the other fields, e.g. we don't explicitly omit the COLOR, NUMBER, or NOTES field if they do not have any content.

**Comment by whalley** [comment]
> I'd really like to see the '[Att]' go from Notes and be replaced by an Attachment column in the transaction list. I'll write a separate enhancement proposal for this at some point so that the pros and cons can be discussed.

@tactilis I can easily change the [Att] to 📎 in output if that would help for now.

**Comment by tactilis** [comment]
> I can easily change the [Att] to 📎 in output if that would help for now.

Thanks for suggesting that, but it's the principle of prepending unrelated information on to the user's Notes that I am concerned about, not whether it is shown as [Att] or📎.

An Attachment is a different attribute of a transaction to a Note and, IMO, the two should not have been combined just for the sake of expediency.

At present, you can't find all the transactions with (or without) attachments by sorting the Notes column to group those having [Att] because, quite correctly, the Notes column sorts on the actual text the user has entered.

![image](https://user-images.githubusercontent.com/4060092/220450387-9d53b57b-08e3-419d-9829-e3c637966292.png)

With a separate Attachment column, which maybe has a paperclip in the header, and a count of the number of attachments (0 = blank, or 1, 2, 3...) as the data, you could easily group transactions with attachments together by sorting that column.
